### DISCUSSION TITLE
关于docker版的alist重启后密码失效的问题

### DISCUSSION THREAD

**Discussion by simmon8888** [AlistGo/alist#3127] [General]
我安装的是Mac版的docker desktop版的Alist，首次进入后通过系统随机密码进去更改了密码后，不重启可以使用，并且添加了存储，然后重启Docker desktop后或者重启机器后再次进入Alist后无法用更改后的秘密进入管理，必须再次使用系统随机的管理密码进去，就是./alist admin 生成的密码，然后所有添加的存储全部被清空，这是怎么回事，需要怎么解决，不能每次重启机器和docker都要更改一次密码和添加一次存储。另外说一下不用Alist不用docker进去管理页面只需更改一次密码和添加一次存储就不会出现这个问题，重启后存储和密码都有效。

**Comment by ifdelete** [comment]


**Reply by chadspring** [reply]
请问如何挂载？

**Reply by yanxile** [reply]
我是挂载出来后，再用./alist admin 就卡在init logrus…不打印了。各种镜像里跟系统上改文件夹权限也没用。

**Comment by 410463305** [comment]
怎么解决的呀？我也是这个问题

**Reply by slyladzhouzhou** [reply]
怎么解决的啊

**Comment by huolanyun** [comment]
我也是 doctor 每次重启后 挂在其他空间的 v3 会失效，需要去后台重启驱动，升到最新版3.21也是一样

**Comment by 410463305** [comment]
我现在用alist helper+alist二进制文件了，不会有重启丢设置的问题了，估计是docker镜像里config文件夹下面的设置文件没有写权限吧？
有个问题是mac在SIP开启的状态下好像不允许陌生程序向外部网络开端口？SIP开启时我只能在本机访问alist的管理页面，其他设备访问都是连接超时，只有关掉SIP后才能在其他电脑上访问

**Reply by slyladzhouzhou** [reply]
怎么设置的 我的这个问题困扰好久了

**Comment by acescofield** [comment]
请问，如何解决该问题，我不是MAC，用威联通部署docker，重启后账户密码被重置，添加的网盘全部丢失

**Comment by slyladzhouzhou** [comment]
解决了么？怎么搞定的？

**Comment by 410463305** [comment]
解决了，不用docker版



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年11月24日(星期五) 上午6:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/alist] 关于docker版的alist重启后密码失效的问题 (Discussion #3127)





 
解决了么？怎么搞定的？

**Reply by slyladzhouzhou** [reply]
不用docker版本用哪个版本

**Comment by slyladzhouzhou** [comment]
不用docker版本alist？那用什么啊

sshock ***@***.***>于2023年11月24日 周五09:14写道：

**Comment by 000ww** [comment]
说一下我的解决方法，我的环境是小米be7000路由器，开启ssh，装的docker。
官方docker一键安装的命令是将alist挂在/opt/alist/data这个目录，重启后这个目录应该是被重置了，要挂在其他目录。我的小米直接挂在/data，因为我的小米路由器docker是装在外置u盘里，是用下面这个命令:
docker run -d --restart=unless-stopped -v alist_data:/data -p 5244:5244 -e PUID=0 -e PGID=0 -e UMASK=022 --name="alist" xhofe/alist:latest

**Reply by sbwcwso** [reply]
> docker run -d --restart=unless-stopped -v alist_data:/data -p 5244:5244 -e PUID=0 -e PGID=0 -e UMASK=022 --name="alist" xhofe/alist:latest

求问这个合集是在 ssh 下运行的吗？我的be7000 提示没有 docker 这个命令。

**Reply by 000ww** [reply]
在 ssh换命令行下看看有没这个   /etc/init.d/mi_docker

**Reply by 000ww** [reply]
实在不行就在docker界面的命令行执行吧

**Comment by zhz-we** [comment]
V3.3 ,每次windows重启，或者docker desktop重启，原来的账号密码就没了，只能重新生成密码
`
随机生成一个密码
docker exec -it alist ./alist admin random

手动设置一个密码,`NEW_PASSWORD`是指你需要设置的密码
docker exec -it alist ./alist admin set NEW_PASSWORD
`
然后之前设置的一些验证都消失了...
总不能每次都要去备份db文件？

**Reply by zhurant** [reply]
请教这个问题怎么解决的

**Reply by zhz-we** [reply]
> 请教这个问题怎么解决的
你了怎么安装的？ docker compose吗？

将data映射到你电脑上的文件夹![image](https://github.com/user-attachments/assets/3d3c7f6c-8107-440b-a425-2011cde8e101)

**Reply by zhurant** [reply]
成功了 感谢🙏  大概是修改 -v 后的文件夹  按照默认语句写的话重启后会初始化

**Comment by CreeperAWA** [comment]
我的Windows也是这样。。。
